regulation-worker/internal/delete/api/api.go

```diff
@@ -62,26 +58,24 @@ func (api *APIManager) Delete(ctx context.Context, job model.Job, destConfig map
 	fileCleaningTime.Start()
 	defer fileCleaningTime.End()
 
-	pkgLogger.Debugf("sending request: %v", req)
 	resp, err := api.Client.Do(req)
 	if err != nil {
-		pkgLogger.Errorf("error while making http request: %v", err)
+		if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
use `os.IsTimeout(err)` and add a test scenario to verify

Reply:
I am not sure if using`os.IsTimeout(err)` would be a good idea. This is what the comment of the func says

> // IsTimeout returns a boolean indicating whether the error is known
> // to report that a timeout occurred.
> //
> // This function predates errors.Is, and the notion of whether an
> // error indicates a timeout can be ambiguous. For example, the Unix
> // error EWOULDBLOCK sometimes indicates a timeout and sometimes does not.
> // New code should use errors.Is with a value appropriate to the call
> // returning the error, such as os.ErrDeadlineExceeded.

Reply:
even they recommend using `errors.Is`

Reply:
then we may use `errors.Is(os.ErrDeadlineExceeded)` but definitely need a test scenario to verify.
If I remember correctly, the http client in Go doesn't return a `context.DeadlineExceeded` if an internal timeout happens.

Reply:
+1 for the test :+1:

Reply:
added.